core/src/main/java/hudson/model/Slave.java

```diff
@@ -162,6 +163,16 @@ public Slave(String name, String nodeDescription, String remoteFS, int numExecut
     	this(name, nodeDescription, remoteFS, numExecutors, mode, labelString, launcher, retentionStrategy, new ArrayList());
     }
 
+    public Slave(@Nonnull String name, String remoteFS, ComputerLauncher launcher) throws FormException, IOException {
```

Comment:
For use from the new `DumbSlave` constructor, though :ant: this one should really be `protected` to clarify that.